### PR TITLE
cherrypicks 1.39

### PR DESCRIPTION
**Pull request by hown3d**
drops 2 commits from previous release:
- https://github.com/gardener/gardener-extension-provider-openstack/commit/e6000ac0af1bceef53704934dc9c1fcf48ca7fed -> arm support is now included in release v1.39
- https://github.com/gardener/gardener-extension-provider-openstack/commit/f906185df2c989f23526a21b42b9369e05e3da4d -> mcm is removed from this repository.

### CODE REVIEW

**Comment by maboehm** [comment]
arent most changes in provider stackit anyways? see some commits of https://github.com/stackitcloud/gardener-extension-provider-stackit/pull/106/commits

since `controlplane` is also part of stackit...
provider-openstack is essentially only used for workers right now.

**Comment by dergeberl** [comment]
Please also add this changes: #52

**Comment by hown3d** [comment]
> Please also add this changes: #52

Needed some adjustments since the healthcheck code changed a bit when updating to 1.39

**Comment by hown3d** [comment]
With the future merge of the openstack specific code e.g. (workers, cloudprovider webhook, healthcheck controller has worker parts and API packages) into the stackit provider it may not be neccessary to upgrade openstack to 1.39.

**Comment by nschad** [comment]
cherry-pick #56

**Comment by MichaelEischer** [comment]
Please squash the `fixup! Cherry-pick useExistingSubnet feature (10-01-2024)` commit

File: charts/gardener-extension-provider-openstack/HEAD [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  values:
+    image:
+      tag: <<<<<<<
```
Comment (MichaelEischer) [inline charts/gardener-extension-provider-openstack/HEAD]: That looks like a massive merge conflict. The `ControllerRegistration` below is completey broken.

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-v1.39.2
+v1.39.2-ske-1
```
Comment (MichaelEischer) [inline VERSION]: Do we really still have to add a suffix? Isn't that already taken care of by the new build process.

File: charts/gardener-extension-provider-openstack/HEAD [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
```
Comment (MichaelEischer) [inline charts/gardener-extension-provider-openstack/HEAD]: What is this file doing here? Judging from the filename the underlying change should have been applied to some other file.

File: VERSION (hunk 2 above)
Comment (Duciwuci) [reply VERSION]: This is an old commit, let me verify the Prow content.

**Comment by Duciwuci** [comment]
@MichaelEischer thanks for feedback. 

However, the pipeline is not working for this extension at version `v1.39`. 

Therefore /hold

**Comment by MichaelEischer** [comment]
@Duciwuci Fixing the build process should be possible by checking whether the script in the vendor folder still exists or switching to the new way to find the gardener path otherwise.

**Comment by Duciwuci** [comment]
> @Duciwuci Fixing the build process should be possible by checking whether the script in the vendor folder still exists or switching to the new way to find the gardener path otherwise.

I'm already on it. You will see a successful run in this PR soon.

**Comment by MichaelEischer** [comment]
/lgtm

Judging from `git range-diff ske-v1.38...cherrypicks-1.39` the cherry-pick is fine.

**Review by nschad — APPROVED** [review]

**Pull request merged**